### PR TITLE
Add squash merge commit title and message options

### PR DESCRIPTION
**Pull request by sfc-gh-kleonhard**
Signed-off-by: Kyle Leonhard <kyle.leonhard@snowflake.com>

## Hey, I just made a Pull Request!

This PR adds options to configure the default squash merge commit title and message.

API references:

* https://github.blog/changelog/2022-08-23-new-options-for-controlling-the-default-commit-message-when-merging-a-pull-request/
* https://octokit.github.io/rest.js/v19
* https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#create-a-repository-for-the-authenticated-user

I didn't see detailed option-level documentation; I'm happy to edit any if it exists!

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/githubRepoCreate.ts [hunk 1]
```diff
@@ -122,6 +126,8 @@ export function createGithubRepoCreateAction(options: {
         deleteBranchOnMerge = false,
         allowMergeCommit = true,
         allowSquashMerge = true,
+        squashMergeCommitTitle = 'PR_TITLE',
+        squashMergeCommitMessage = 'COMMIT_MESSAGES',
```
Comment (agentbellnorm) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/githubRepoCreate.ts:130]: Is this a new default behavior, or is this how it was before? If it's new I think we can mention it in the changelog.

**Review by agentbellnorm — COMMENTED** [review]
Thanks for the PR! It looks good to me. I had one question/comment.

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/githubRepoCreate.ts (hunk 1 above)
Comment (sfc-gh-kleonhard) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/githubRepoCreate.ts:130]: Good eye! I thought I'd matched the default, but a careful read of materials ([blog post](https://github.blog/changelog/2022-08-23-new-options-for-controlling-the-default-commit-message-when-merging-a-pull-request/) tells me the default is "default message," described as:

> Default message: commit title and commit message (if the pull request contains a single commit), or the pull request title and number and list of commits (if the pull request contains multiple commits)

This tells me the default commit title is COMMIT_OR_PR_TITLE and the message is COMMIT_MESSAGES.

I'll change things. Alternatively, being that this field is optional, I could leave the default undefined and rely on the APIs defaults. I didn't see prior art for this in other parameters though. Let me know what backstage prefers in these circumstances.

Comment (sfc-gh-kleonhard) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/githubRepoCreate.ts]: Updated the default to COMMIT_OR_PR_TITLE.

Again, let me know if I should leave the default as undefined and rely on the underlying APIs defaults instead.

**Comment by sfc-gh-kleonhard** [comment]
Bump, what do you think?

Alternatively, we could leave the defaults for these options undefined effectively delegating defaults to the API. I did this in another [PR](https://github.com/backstage/backstage/pull/15291/files#diff-40c97a2355793782b547f90cf4e252e82675c2388ee878f258b9d344a393c41e) for a few other options

**Review by Rugvip — APPROVED** [review]
Alright, thank you! 👍

**Pull request merged**